[Docs] Add info on logs on Kubernetes & metrics in the ops section

realvictorprm opened the pull request:
@cocreature Is this what you had roughly in mind regarding documentation? Happy to recieve more input on this.
The current docs look like this if deployed locally.

![image](https://user-images.githubusercontent.com/25414896/128735672-0b755246-a74a-42d3-b8f4-c54ae3696719.png)

changelog_begin

- [Docs] Information was added in the `Operating Daml` section on how to aggregate logs on Kubernetes in conjuction with Daml services & what options exists for exporting metrics from daml services (not Kubernetes specific)

changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

cocreature commented:
> @cocreature Is this what you had roughly in mind regarding documentation? Happy to recieve more input on this.
> The current docs look like this if deployed locally.

I’d suggset to add a "Operating Daml Connect" subsection and put this stuff under there so we get somewhat sensible grouping.

victormueller-da commented:
Hm good catch, `Operating Daml` is near but not `Operating Daml Connect`. Will move it :)

realvictorprm commented:
Moved, now it looks like this. I moved the new section to the end because 1. it's new and 2. having the new section next to the `Operating Daml` one looks confusing first (at least in my opinion).
![image](https://user-images.githubusercontent.com/25414896/128852056-cea89665-0ef7-4e45-af8d-7c5a78078920.png)

cocreature commented:
I was more thinking about having `Operating Daml Connect` be a subsection of `Operating Daml` but I don’t feel strongly about this. @bame-da do you have an opinion here?

realvictorprm commented:
I think it would be even more confusing to have it as subsection of `Operating Daml`. Maybe the section should be overall called differently i.e. `Operating on the Cloud`. Geez, hard to come up with a better name ._.

cocreature commented:
A general "Operating Daml" section with subsections for "Operating Daml Connect", "Operating Daml on SQL", … doesn’t seem that confusing to me.

victormueller-da commented:
> A general "Operating Daml" section with subsections for "Operating Daml Connect", "Operating Daml on SQL", … doesn’t seem that confusing to me.

If you put it like that, then yes it makes sense to me.

victormueller-da commented:
will move it again then :smile:

realvictorprm commented:
Ok, moved and now it looks like this:
![image](https://user-images.githubusercontent.com/25414896/128861009-51ada4d7-20b6-4aee-b0ac-98b895ccb8e8.png)

that good?

cocreature commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,27 @@
+To collect all logs for analyzing them e.g. in your `ElasticSearch` instance, one easy way to achieve this is using `Filebeat <https://www.elastic.co/beats/filebeat>`_.
+It regulary collects the logs of your containers and ingests them into `Elasticsearch`, `Logstash <https://www.elastic.co/logstash/>`_ or `Kafka <https://kafka.apache.org/>`_ etc. and the best is, it doesn't require any additional configuration for the daml services.
+
+You can find external documentation on, how to setup `ElasticSearch` with `Filebeat` and `Kibana` for analyzing logs on your Kubernetes cluster `here <https://www.deepnetwork.com/blog/2020/01/27/ELK-stack-filebeat-k8s-deployment.html>`_.
```
We should turn this into a full section that shows a fully working setup and doesn’t just link to an external blogpost.

realvictorprm replied:
I can do that, however to what extend do we want to do it here?
E.g. my first thought would be to make it like a blog post where one shows how to create that setup on a local minikube cluster.

cocreature replied:
Basing it on a minicube example sounds sensible :+1:

cocreature commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,27 @@
+
+You can find external documentation on, how to setup `ElasticSearch` with `Filebeat` and `Kibana` for analyzing logs on your Kubernetes cluster `here <https://www.deepnetwork.com/blog/2020/01/27/ELK-stack-filebeat-k8s-deployment.html>`_.
+
+Daml Metrics options
```
Similar here, this should show an example of integrating this with prometheus and not just list CLI flags.

realvictorprm replied:
Same as for the other, Minikube cluster setup?

realvictorprm replied:
Side note, I would like to keep these sections and add an additional example section because I think it is good to have an overview + example.

realvictorprm replied:
Great, many thanks for the input. Then I can proceed further here :)

realvictorprm commented:
I added instructions on how to setup a cluster such that all what is required for the logging and metrics can be applied.

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+Log aggregation on Kubernetes
+=============================
+
+If configured so, every daml service can output its logs in JSON format.
```
```suggestion
If configured so, every daml service can output its logs in JSON format on stdout.
```

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+Prerequisites
+--------------
+
+For this to work, you need to have `docker` & `minikube` installed (If you run into any issue with the kubernetes proxy and minikube on Ubuntu, this is the fix [https://github.com/kubernetes-sigs/kind/issues/2240#issuecomment-888213835]).
```
Does that work as a link? I've never seen that syntax in RST documentation.

garyverhaegen-da replied:
Maybe:
```suggestion
For this to work, you need to have `docker` & `minikube` installed (If you run into any issue with the kubernetes proxy and minikube on Ubuntu, `this fix may be worth trying <https://github.com/kubernetes-sigs/kind/issues/2240#issuecomment-888213835>`_).
```

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+
+For this to work, you need to have `docker` & `minikube` installed (If you run into any issue with the kubernetes proxy and minikube on Ubuntu, this is the fix [https://github.com/kubernetes-sigs/kind/issues/2240#issuecomment-888213835]).
+
+Installing elasticsearch on your local kubernetes cluster
```
```suggestion
Installing Elasticsearch On Your Local Kubernetes Cluster
```

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+
+- console output (logs) via ``--metrics-reporter console``
+
+Deployment example with minikube
```
```suggestion
Deployment Example With Minikube
```

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+To collect all logs for analyzing them e.g. in your `ElasticSearch` instance, one easy way to achieve this is using `Filebeat <https://www.elastic.co/beats/filebeat>`_.
+It regulary collects the logs of your containers and ingests them into `Elasticsearch`, `Logstash <https://www.elastic.co/logstash/>`_ or `Kafka <https://kafka.apache.org/>`_ etc. and the best is, it doesn't require any additional configuration for the daml services.
+
+Daml Metrics options
```
```suggestion
Daml Metrics Options
```

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+Installing elasticsearch on your local kubernetes cluster
+=========================================================
+
+First you need to add the elastic repository to helm, so the chart can be found.
```
```suggestion
First you need to add the Elastic repository to helm, so the chart can be found.
```

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+helm repo update
+``
+
+Now you can setup the elastic search cluster via:
```
```suggestion
Now you can setup the Elasticsearch cluster via:
```

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+helm upgrade --install es --version 7.9.3 elastic/elasticsearch --values https://raw.githubusercontent.com/elastic/helm-charts/master/elasticsearch/examples/minikube/values.yaml
+``
+
+Side note: Please do not use this config setup in production as antiAffinity is set to soft, which effectively allows elasticsearch to run all pods on one physical machine. However for local testing with minikube this is necessary and thus fine. 
```
```suggestion
Side note: Please do not use this config setup in production as ``antiAffinity`` is set to ``soft``, which effectively allows Elasticsearch to run all pods on one physical machine. However for local testing with minikube this is necessary and thus fine. 
```

garyverhaegen-da commented on docs/source/ops/connect/content.rst:
```diff
@@ -0,0 +1,121 @@
+
+.. _ops-connect-content_index:
+
+Log aggregation on Kubernetes
```
```suggestion
Log Aggregation On Kubernetes
```

garyverhaegen-da commented on docs/source/ops/connect/index.rst:
```diff
@@ -0,0 +1,13 @@
+.. toctree::
+   :maxdepth: 3
+
+   content
```
maybe rename that file to `log-and-metrics` or something? As we add further sections, having one called `content` is going to be confusing.

realvictorprm replied:
I agree, will do
